assets/src/blocks/Spreadsheet/SpreadsheetFrontend.js

```diff
@@ -0,0 +1,217 @@
+            loading: false,
+            spreadSheetData
+          } );
+      } );
+    } else {
+      if ( this.props.handleErrors ) {
+        this.props.handleErrors( { invalidSheetId: true } );
+      }
+      this.setState({ loading: false, spreadSheetData: null });
+    }
+  }
+
+  componentDidMount() {
+    if ( this.props.url != '' ) {
+      this.fetchSheetData(this.props.url);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.url !== this.props.url) {
+      this.fetchSheetData(nextProps.url);
+    }
+  }
+
+  extractSheetID( url ) {
+    const googleSheetsPattern = /https:\/\/docs\.google\.com\/spreadsheets\/d\/e\/([\w-]+)/;
+    const matches = url.match(googleSheetsPattern);
+    if (matches !== null) {
+      return matches[1];
+    } else {
+      return false;
+    }
+  }
+
+  sortRows( rows, columnIndex ) {
+    if (columnIndex === null) {
+      return rows;
+    }
+
+    const sortedRows = rows.sort( ( rowA, rowB ) => {
+      const textCompare = rowA[columnIndex].localeCompare( rowB[columnIndex] );
+      if ( textCompare !== 0 ) {
+        return textCompare;
+      }
+    } );
+    if (this.state.sortDirection === 'desc') {
+      return sortedRows.reverse();
+    } else {
+      return sortedRows;
+    }
+  };
+
+  filterMatchingRows( rows ) {
+    const filteredRows = rows.filter( row => {
+      return !!row.find( cell => cell.toLowerCase().includes( this.state.searchText.trim().toLowerCase() ) );
+    })
+    return filteredRows;
+  }
+
+  getRows() {
+    if ( this.state.spreadSheetData === null ) {
+      return placeholderData.rows;
+    } else if ( this.state.loading === true || this.state.loading === null ) {
+      return [];
+    } else {
+      return this.state.spreadSheetData.rows;
+    }
+  }
+
+  renderRows() {
+    const rows = this.sortRows( this.filterMatchingRows( this.getRows() ), this.state.sortColumnIndex );
+
+    return this.state.searchText.length > 1 && rows.length === 0
+      ? <tr>
+          <td colSpan="99999">
+            <div className='spreadsheet-empty-message'>
+              { __('No data matching your search.', 'planet4-blocks') }
+            </div>
+          </td>
+        </tr>
+      : rows.map((row, rowNumber) => (
+          <tr key={ rowNumber } data-order={ rowNumber }>
+            {
+              row.map((cell, cellIndex) => (
+                <td key={ cellIndex }>
+                  {
+                    this.state.searchText.length > 0
+                    ? HighlightMatches( cell, this.state.searchText )
+                    : cell
+                  }
+                </td>
+              ))
+            }
+          </tr>
+        ));
+  }
+
+  render() {
+    const headers = this.state.spreadSheetData
+      ? this.state.spreadSheetData.header
+      : placeholderData.header;
+
+    return (
+      <Fragment>
+        <div className="block-spreadsheet" style={{ cssText: CSSVariablesToInlineText( this.props.css_variables ) }}>
```

Comment:
do you think it would be too complex to have the block inside a `<section>`, instead of a `<div>`. To be more semantically meaningful, but also to preserve consistency with existing blocks markup.

Reply:
Sure, forgot about that.